### DISCUSSION TITLE
Lesson 9, Part 28 - Index Out of Bounds Error on testPerformUpKeepUpdatesRaffleAndEmitsRequest (using Vm.log)

### DISCUSSION THREAD

**Discussion by 0xwtfwtf** [Cyfrin/foundry-full-course-cu#673] [General]
Why am I getting an index out of bound error on this test (see screen cap with my code and terminal below)? I have the same code as Patrick. I imported Vm.

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/138697105/fd9d1e8e-92cb-465b-bb19-6eae35fa9739)

**Comment by shikhar229169** [comment]
Have you emitted the event in performUpkeep with requestId as indexed parameter.

**Comment by theirrationalone** [comment]
In your stack logs.... 

there's a log `Vm::getRecordedLogs();` which is telling that you have only one entry in which you have an array which contains your event data on there on same entry's array at index 2, You have your `requestId` maybe it is Or not???? You might forgot to emit that same event for your test. Please copy that same event into your test for very specific logs at entry index 1.
